packages/pubsub/src/subscription.js

```diff
@@ -592,157 +442,398 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
-
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
-  };
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
   var reqOpts = {
     subscription: this.name
   };
 
-  this.parent.request(protoOpts, reqOpts, function(err, resp) {
-    if (err) {
-      callback(err, resp);
-      return;
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.removeAllListeners('message');
```

Comment:
This part?

> it doesn't make sense to automatically remove error listeners since that is the only way we can inform the user that an error occurred.

It seems like the user can be in a state where they have a Subscription object that is waiting for an "error" event that cannot come. If the subscription was deleted successfully, what can error? If we don't remove listeners, the object can't be freed from memory.